WooCommerce/Classes/POS/Models/Cart.swift

```diff
@@ -49,7 +49,9 @@ extension Cart {
         case .variableParentProduct:
             return
         case .coupon(let coupon):
-            let couponItem = Cart.CouponItem(id: UUID(), code: coupon.code, summary: coupon.summary)
+            guard !coupons.contains(where: { $0.id == coupon.id }) else { return }
```

Comment:
Do we need this check here if we skip the duplicate in the action handler?

Reply:
We don't anymore, I've removed this on https://github.com/woocommerce/woocommerce-ios/pull/15551/commits/68096bc0ddf9c9ddc9efa60df3e378fe25f89757 and added tests for the action handler on https://github.com/woocommerce/woocommerce-ios/pull/15551/commits/a7c59776e0c0033bfb34bdb4f9be5f1b9dae34e1